modules/social_features/social_content_block/social_content_block.module

```diff
@@ -25,56 +28,136 @@ function social_content_block_preprocess_block(&$variables) {
-    }
+function _social_content_block_get_content_type_for_block(BlockContentInterface $block_entity) {
+  if ($block_entity->field_plugin_id->isEmpty()) {
+    return '';
+  }
+  $plugin_id = $block_entity->field_plugin_id->value;
+  $definition = \Drupal::service('plugin.manager.content_block')->getDefinition($plugin_id);
+
+  // For nodes we distinguish bundles, otherwise we only show the entity type.
+  if ($definition['entityTypeId'] === 'node') {
+    return $definition['bundle'];
   }
-  return FALSE;
+
+  return $definition['entity_type'];
 }
 
 /**
- * Implements hook_form_BASE_FORM_ID_alter().
+ * Implements hook_form_alter().
  */
-function social_content_block_form_block_content_custom_content_list_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  // Add general description above first field to explain what the filters do.
-  $form['field_topic_type'] += [
-    '#prefix' => t('To make the list of topics more specific you can additionally configure more filters such as topic types, content tags and groups.'),
+function social_content_block_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form_ids = [
+    'block_content_custom_content_list_form',
+    'block_content_custom_content_list_edit_form',
   ];
 
-  // Add description per field explaining what to fill in.
-  $form['field_topic_type'] += [
-    '#suffix' => t('Autocomplete field with items from taxonomy list topic types.'),
-  ];
-  $form['field_content_tags'] += [
-    '#suffix' => t('Autocomplete field with items from taxonomy list content tags.'),
-  ];
-  $form['field_group'] += [
-    '#suffix' => t('Autocomplete field with group names.'),
-  ];
+  if (!in_array($form_id, $form_ids)) {
+    return;
+  }
 
+  /** @var \Drupal\social_content_block\ContentBlockManagerInterface $content_block_manager */
+  $content_block_manager = \Drupal::service('plugin.manager.content_block');
+
+  foreach ($content_block_manager->getDefinitions() as $plugin_id => $plugin_definition) {
+    $fields = &$form['field_plugin_field']['widget'][0][$plugin_id]['#options'];
+
+    foreach ($fields as $field_name => $field_title) {
+      if ($field_name === $field_title) {
+        $fields[$field_name] = $form[$field_name]['widget']['target_id']['#title'];
+
+        $form[$field_name]['#states'] = [
+          'visible' => [
+            ':input[name="field_plugin_id[0][value]"]' => [
+              'value' => $plugin_id,
+            ],
+            ':input[name="field_plugin_field[0][' . $plugin_id . ']"]' => [
+              ['value' => 'all'],
+              ['value' => $field_name],
+            ],
+          ],
+        ];
+      }
+    }
+  }
+
+  $form['#attributes']['class'][] = 'form--default';
+
+  foreach (Element::children($form['actions']) as $key) {
+    if ($key !== 'submit') {
+      $form['actions'][$key]['#attributes']['class'][] = 'btn';
+      $form['actions'][$key]['#attributes']['class'][] = 'btn-flat';
+    }
+  }
+}
```

Comment:
I know it's annoying but this should happen in one of the themes.

Reply:
Done